app/scripts/views/mixins/settings-panel-mixin.js

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// helper functions for views with a profile image. Meant to be mixed into views.
+
+define([
+  'jquery',
+  'views/base'
+], function ($, BaseView) {
+  'use strict';
+
+  return {
+    initialize: function (options) {
+      this.superView = options.superView;
+    },
+    events: {
+      'click .settings-unit-toggle': BaseView.preventDefaultThen('_triggerPanel'),
+      'click .cancel': BaseView.preventDefaultThen('closePanelReturnToSettings')
+    },
+
+    _triggerPanel: function (event) {
+      var href = event && $(event.currentTarget).data('href');
+      if (href) {
+        this.navigate(href);
+      }
+    },
+
+    openPanel: function () {
+      var unit = this.$('.settings-unit');
+      unit.addClass('open');
+    },
+
+    closePanelReturnToSettings: function () {
+      this.navigate('settings');
+      this.closePanel();
+    },
+
+    closePanel: function () {
+      this.$('.settings-unit').removeClass('open');
+    },
+
+    displaySuccess: function (msg) {
+      var self = this;
+      if (! self.superView) {
+        return;
+      }
+      self.superView.displaySuccess(msg);
+      self.closePanel();
```

Comment:
It took me the entire bike ride to work to talk myself into agreeing with both arguments. The advantage to the current approach is that the behavior is only added if the mixin is added, and the user of a mixin has the responsibility to understand its side effects. It's an AOP solution that a developer has to add to their view. Remove the mixin, remove the overridden functionality.

The downside is for developers who don't understand the side effects of the mixin they use. A second downside is a developer has to call `BaseView.prototype.displaySuccess.call(this` if they want to display an in-panel success message.

I've talked myself into your point of view. Can you document the mixin behavior w.r.t. displaySuccess?